clang-tools-extra/docs/clang-tidy/checks/bugprone/tagged-union-member-count.rst

```diff
@@ -0,0 +1,241 @@
+.. title:: clang-tidy - bugprone-tagged-union-member-count
+
+==================================
+bugprone-tagged-union-member-count
+==================================
+
+Gives warnings for tagged unions, where the number of tags is
+different from the number of data members inside the union.
+
+A struct or a class is considered to be a tagged union if it has
+exactly one union data member and exactly one enum data member and
+any number of other data members that are neither unions or enums.
+
+Example
+=======
```

Comment:
```suggestion
-------
```

(Although the example(s) are usually not individual heading levels in the Tidy documentation.)